scripts/utils.js

```diff
@@ -0,0 +1,60 @@
+import { validators } from "./validators.js";
+/**
+ * debounce code
+ */
+export function debounce(fn, delay = 300) {
+  let timer;
+  return function (...args) {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+      fn.apply(this, args);
+    }, delay);
+  };
+};
+/**
+ *  toggle password
+ */
+export const togglePasswordVisibility = (button) => {
+  const targetId = button.getAttribute("data-target");
+  const passwordInput = document.getElementById(targetId);
+  if (!passwordInput) return;
+  const img = button.querySelector("img");
+  if (passwordInput.type === "password") {
+    passwordInput.type = "text";
+    img.src = "/images/btn_visibility_on_24px.png";
+    img.alt = "비밀번호 보이는 중";
+  } else {
+    passwordInput.type = "password";
+    img.src = "/images/btn_none_visibility_on_24px.png";
+    img.alt = "비밀번호 숨겨진 상태";
+  }
+};
+/*
+ *  Validate User Input
+ */
+export const validateInputs = (inputEl) => {
+  const validateType = inputEl.dataset.validate;
+  const { value, id } = inputEl;
+  if (!validateType || !validators[validateType]) return;
+  const { isValid, message } = validators[validateType](value);
+  const errMsg = document.getElementById(`${id}-error`);
+  if (errMsg) {
+    errMsg.textContent = isValid ? "" : message;
+  }
+  if (isValid) {
+    inputEl.classList.remove("input-error");
+  } else {
+    inputEl.classList.add("input-error");
+  }
```

Comment:
이 조건문도 classList.toggle을 사용하면 이렇게 간소화할수있답니다!

[참고](https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/toggle)

```suggestion
inputEl.classList.toggle("input-error", !isValid);
```